serverless/internal/layout/README.md

```diff
@@ -0,0 +1,86 @@
+Layout
+======
+
+This doc describes the on-disk layout and format of the serverless log files.
+
+> :warning: this is still under development, and the exact layout and contents
+> of the serverless log structure may well change.
+
+Broadly, the layout consists of a self-contained directory hierarchy whose
+contents represents the entire state of the log. With one exception, the
+contents and structure of this directory is designed to allow it to be safely
+and indefinitely cached by clients.
+
+Inside the directory you'll find:
+
+ * :file_folder: leaves/
+ * :file_folder: seq/
+ * :file_folder: tile/
+ * :page_facing_up: state
+
+
+state
+-----
+`state` contains a JSON formatted [`LogState struct`](../../api/state.go) that
+represents the latest state of the log tree.
+This is the *only* file in the serverless log data set which *should not* be
+indefinitely cached by serving infrastructure or clients.
+
+seq/
+----
+`seq/` contains a directory hierarchy containing leaf data for each sequenced
+entry in the log.
+
+To avoid creating a large directory containing many files, the following scheme
+is used: the sequence number is interpreted as a 48 bit number, e.g.
+`0x123456789a`, and a prefix directory hierarchy is created from that like so:
+`.../seq/12/34/56/78/9a`.
+
+leaves/
+-------
+`leaves/` contains files which map all known leaf hashes to their position in
+the log.
+
+Similarly to `seq/`, a prefix directory scheme is used: given a leaf hash
+`0x0123456789ABCDEF0000000000000000`, the file used to store this leaf's index
+in the log would be: `.../leaves/01/23/45/6789abcdef0000000000000000`.
+
+The contents of the file is simply the hex ASCII string representation of the
+index.
+
+tile/
+-----
+`tile/` contains the internal nodes of the log tree.
+
+The nodes are grouped into "tiles" - these are 8 level deep "sub trees".
+Each tile exists at a particular location in tile-space, identified by its
+`layer` and `index`.
```

Comment:
Consider strata instead of layer. It would be consistent with the map terminology.

Reply:
changed